torchinductor/debug.py

```diff
@@ -138,3 +169,148 @@ def in_output(snode):
 
     graph.output(outputs[0] if len(outputs) == 1 else tuple(outputs))
     return graph
+
+
+class DebugContext:
+    _counter = itertools.count()
+
+    @staticmethod
+    def wrap(fn):
+        @functools.wraps(fn)
+        def inner(*args, **kwargs):
+            with DebugContext():
+                return fn(*args, **kwargs)
+
+        return wrap_debug(inner, compiler_name="inductor")
+
+    @staticmethod
+    def create_debug_dir():
+        for n in DebugContext._counter:
+            dirname = os.path.join(cache_dir(), f"debug.{os.getpid()}.{n}")
```

Comment:
Maybe make this a config option and/or env var? I'd imagine this would be useful to control per run without modifying cache_dir

Reply:
This directory is only a temporary one, so the name doesn't matter.  We rename this later to sit right beside the output code.  So if we write `foo123.py` this will be named `foo123.debug/`